libbpfgo.go

```diff
@@ -3,77 +3,80 @@ package libbpfgo
 /*
 #cgo LDFLAGS: -lelf -lz
 
-#include <bpf/bpf.h>
-#include <bpf/libbpf.h>
-#include <stdint.h>
-#include <stdio.h>
+#include <errno.h>
+#include <fcntl.h>
 #include <stdlib.h>
 #include <sys/resource.h>
 
-#include <asm-generic/unistd.h>
-#include <errno.h>
-#include <fcntl.h>
-#include <linux/perf_event.h>
-#include <linux/unistd.h>
-#include <string.h>
-#include <unistd.h>
+#include <bpf/bpf.h>
+#include <bpf/libbpf.h>
 
 #ifndef MAX_ERRNO
 #define MAX_ERRNO       4095
 
 #define IS_ERR_VALUE(x) ((x) >= (unsigned long)-MAX_ERRNO)
 
-static inline bool IS_ERR(const void *ptr) {
-	return IS_ERR_VALUE((unsigned long)ptr);
+static inline bool IS_ERR(const void *ptr)
+{
```

Comment:
Why changing to this convention?
In our code we keep the opening brace in the function signature line

Reply:
Just so you don't think I didn't put some thoughts on this before suggesting:

This is K&R style ([kernel variant](https://en.wikipedia.org/wiki/Indentation_style#Variant:_Linux_kernel)), which I think we should start complying to in our C sources.

Reason: With K&R style we can check formatting of this, and our eBPF source codes, with tools such as check_patch, during the PRs (Not particularly on this case, but, this code should also be extracted from .go code if possible (near future) so we can check formatting in PRs).

In this case, since it is part of golang src file, it has tabs expanded to 4 chars, which shouldn't be true for a C src file, but doesn't make sense to have different tab expansion policy in a go file.

If you don't like it, we can move back to previous style (with fixes to mixed tab and space indendation it had here and there). I just consider an error treating C code with a go formatting pattern (as well as having +100 cols lines in golang src, which is something we should try to avoid to improve code readability).

WDYT ?

But I'd like you to request for you to consider K&R kernel style not for this, but also for the eBPF code refactoring within tracee.

Reply:
Ok, that's a good cause.
I'm ok with this change